src/core/context/index.ts

```diff
@@ -97,6 +97,17 @@ export class Context implements AbstractContext {
   }
 
   public updateEvent(path: string, val: unknown): SegmentEvent {
+    // Allow integrations that are set to true to be overridden with integration settings.
+    // For example, allow {..., integrations: {Amplitude: true}} to be overwritten with
+    // {..., integrations: {Amplitude: {sessionId: 'foo'}}}.
+    if (path.split('.')[0] === 'integrations') {
+      const integrationName = path.split('.')[1]
+
+      if (this._event.integrations?.[integrationName] === true) {
+        delete this._event.integrations?.[integrationName]
```

Comment:
🤔 one quick thing to check here. Looks like the [latest version ](https://github.com/lukeed/dset/releases/tag/v3.0.0) of `dset` now allows overriding existing non-object values. So in this case we don't have to delete the nested value before calling `dset`.

Reply:
Can we check if upgrading `dset` works here without breaking any of the existing tests?